### DISCUSSION TITLE
k8gb behind a NAT with cloudflare

### DISCUSSION THREAD

**Discussion by suppersam** [k8gb-io/k8gb#1668] [Q&A] [answered]
Hi 
I have an nginx ingress controller service type load balancer that receives an ip from the metallb service and at the firewall level i create a NAT pointing 80/443 to this public IP. When i create a new Gslb / ingress the ingress gets the local IP and creates a type A record on CloudFlare pointing to the local IP instead of a Public IP. The question is, can i use a static public IP on a Gslb object so ClouFlare can receive and point to the correct public IP address?

**Comment by ytsarev** [comment]
Hey @suppersam, thanks a lot for trying out the project! 

We are retrieving the IP address for A record from the exposed coredns service or the Ingress address with the host network. Please see the detailed explanation here: https://github.com/k8gb-io/k8gb/discussions/364#discussioncomment-9868944.

There is no ability to override the IP address that is not coming from any controller at the moment.

It looks like a valid case to cover so please feel free to write a detailed Issue with the feature request 👍

**Comment by Aksine** [comment]
Hey Sam, i believe this feature is similar to this issue #1275 and it's been added to the roadmap.

**Comment by ytsarev** [comment, marked as answer]
https://github.com/k8gb-io/k8gb/issues/1275 was implemented and released
